constructor/conda_interface.py

```
@@ -87,20 +91,41 @@ def get_repodata(url):
 
         return full_repodata
 
-    def write_repodata(cache_dir, url, full_repodata, used_packages):
+    def write_repodata(cache_dir, url, full_repodata, used_packages, info):
         used_repodata = {k: full_repodata[k] for k in
                          set(full_repodata.keys()) - {'packages', 'packages.conda', 'removed'}}
-        repodata_filename = _cache_fn_url(used_repodata['_url'].rstrip("/"))
         used_repodata['packages.conda'] = {}
         used_repodata['removed'] = []
         # arbitrary old, expired date, so that conda will want to immediately update it
         # when not being run in offline mode
         used_repodata['_mod'] = "Mon, 07 Jan 2019 15:22:15 GMT"
         used_repodata['packages'] = {
             k: v for k, v in full_repodata['packages'].items() if v['name'] in NAV_APPS}
+
+        # Minify the included repodata
         for package in used_packages:
-            for key in ('packages', 'packages.conda'):
-                if package in full_repodata.get(key, {}):
-                    used_repodata[key][package] = full_repodata[key][package]
+            key = 'packages.conda' if package.endswith(".conda") else 'packages'
+            if package in full_repodata.get(key, {}):
+                used_repodata[key][package] = full_repodata[key][package]
+                continue
+            # If we're transcoding packages, fix-up the metadata
+            if package.endswith(".conda"):
+                original_package = package[:-len(".conda")] + ".tar.bz2"
+                original_key = "packages"
+            elif package.endswith(".tar.bz2"):
+                original_package = package[:-len(".tar.bz2")] + ".conda"
+                original_key = "packages.conda"
+            else:
+                print("WARNING: package type is unknown for: %s" % package)
+                continue
```

Shouldn't this fail loudly since that implies the metadata is corrupt?

---

> I'm in favour of that, how about:
> 
> ```diff
> -                print("WARNING: package type is unknown for: %s" % package)
> -                continue
> +                raise NotImplementedError("Package type is unknown for: %s" % package)
> ```

Looks good to me!